### PR TITLE
#1 - WIP commit for code feedback + collaboration

### PR DESCRIPTION
**Pull request by adupuis2**
It was discussed that @alexinman might want to build a new db schema for this project.

While awaiting that schema rework, I wanted to get some feedback. I believe this is safe to merge in currently.

### CODE REVIEW

File: app/controllers/elo_controller.rb [hunk 1]
```diff
@@ -189,6 +191,25 @@ def game
     end
   end
 
+  def challenge
+    reply "Hey, we're working on it!" and return # safeguard while wip
+
+    _, p1, p2, type, p3 = params[:text].match(/^#{SLACK_ID_REGEX}(?: +and +#{SLACK_ID_REGEX})? +at +(.*)(?: +with +#{SLACK_ID_REGEX})?$/).to_a
```
Comment (adupuis2) [inline app/controllers/elo_controller.rb]: I am not sure this is the best way to handle challenging in doubles.
`/challenge with @teammate @opponent1 @opponent2 at @gametype` seems bad too.
`/challenge @opponent1 and @opponent2 with @teammate at @gametype` also seems bad.

🤷‍♀

File: db/schema.rb [hunk 2]
```diff
@@ -42,7 +42,19 @@
     t.integer  "team_size",    default: 1,    null: false
   end
 
+  create_table  "challenges",                 force: :cascade do |t|
+    t.integer   "challenge_id",               null: false
```
Comment (adupuis2) [inline db/schema.rb]: In the tables @alexinman created, the fks are a mix of `text` and `integer`.

Why?

File: db/schema.rb [hunk 3]
```diff
@@ -42,7 +42,18 @@
     t.integer  "team_size",    default: 1,    null: false
   end
 
+  create_table  "challenges",                 force: :cascade do |t|
+    t.integer   "player_one_id",              null: false
```
Comment (adupuis2) [inline db/schema.rb]: In the tables @alexinman created, the fks are a mix of `text` and `integer`.

Why?

File: db/schema.rb [hunk 4]
```diff
@@ -42,7 +42,18 @@
...
+    t.integer   "player_two_id",              null: false
+    t.integer   "game_id",     default: -1,   null: false
```
Comment (adupuis2) [inline db/schema.rb]: Would it be better to permit this to be null and default to null? I figured the fewer `null`s going around, the better.

File: db/schema.rb [hunk 5]
```diff
@@ -42,7 +42,18 @@
...
+    t.datetime  "created_at",                 null: false
+    t.datetime  "updated_at",                 null: false
+  end
+
   add_foreign_key "games", "players", column: "player_one_id", on_delete: :cascade
   add_foreign_key "games", "players", column: "player_two_id", on_delete: :cascade
   add_foreign_key "players", "game_types", on_delete: :cascade
+  add_foreign_key "challenges", "players", column: "player_one_id", on_delete: :cascade
```
Comment (adupuis2) [inline db/schema.rb]: Teams of players are entered as a single `Player` in the `Player` table, right?

File: db/schema.rb (hunk 4 above)
Comment (alexinman) [reply db/schema.rb]: Yes, I agree. A `null` value here would indicate that the game hasn't been logged yet or the challenge was declined.

File: db/schema.rb (hunk 5 above)
Comment (alexinman) [reply db/schema.rb]: Correct.

File: db/schema.rb (hunk 3 above)
Comment (alexinman) [reply db/schema.rb]: I'm not sure what you're talking about. The 3 fks that I've added are all integers.

File: db/schema.rb (hunk 2 above)
Comment (alexinman) [reply db/schema.rb]: [See this comment.](https://github.com/alexinman/elo-slack-app/pull/6#discussion_r294912179)

File: db/schema.rb (hunk 3 above)
Comment (alaindupuis-introhive) [reply db/schema.rb]: Disregard. 
image
I took these text fields as fks, but now I realize they are literally strings in the format of "@username"

Comment (alexinman) [reply db/schema.rb]: They're actually the public Slack User Ids. Example: `"@UJ9AJH49E"`
For teams that have more than one player, it's the public Slack User Ids of all team members sorted and joined with dashes. Example: `"@UEBBTBWLQ-@UJ9AJH49E"`

I guess that should have been clearer, but I'm too lazy to change it now.

File: app/controllers/elo_controller.rb (hunk 1 above)
Comment (alexinman) [reply app/controllers/elo_controller.rb]: Hmm.. not sure about this one either. Maybe we can ask around in Slack and see what other people think about it and get their input. I think I prefer the latter option more, but it's still not super nice.

File: app/models/player.rb [hunk 6]
```diff
@@ -19,6 +19,10 @@ def tied_with(other_player, logged_by_user_id)
     log_game(other_player, logged_by_user_id, :plays_draw)
   end
 
+  def challenged(other_player)
+    # wip pending updates to schema
```
Comment (alexinman) [inline app/models/player.rb]: I'd rather we get this complete before merging it in.

File: app/controllers/elo_controller.rb [hunk 7]
```diff
@@ -23,6 +23,8 @@ def elo
       games
     when "help"
       help
+    when "challenge"
+      challenge
```
Comment (alexinman) [inline app/controllers/elo_controller.rb]: I think it would make sense to create a new controller `ChallengeController` to deal with challenges.

**Pull request closed without merging**